packages/node-resolve/test/package-entry-points.js

```diff
@@ -352,7 +352,7 @@ test('can override a star pattern using null', async (t) => {
 
 test('can self-import a package when using exports field', async (t) => {
   const bundle = await rollup({
-    input: 'self-package-import',
+    input: 'self-package-import.js',
```

Comment:
This is a tricky one from an extensions perspective, because it's unclear if we are building a package or a path so there is some ambiguity in place.

Do we try the package lookup first, then fall back to a path-like lookup? Even still, a path-like lookup should still support extension checking I think.

Reply:
With `self-package-import` we get
```
Could not resolve import "self-package-import" in undefined using exports defined in /Users/tomjenkinson/Documents/GitHub/plugins/packages/node-resolve/test/fixtures/node_modules/self-package-import/package.json.
```
because it does try treating it as the package, and does not fall back then to `self-package-import.js` given the package had an export map.

Are you saying we should fall back to trying the candidates with extensions if it's the root import (no importer)?

Reply:
I see, so because it is matching the package lookup path, we don't try the path-like resolution.

That makes sense to me then.

Reply:
IMO we shouldn't have a special case for the root import, unless there is a good use case for it?

The probability of users having a file named the same as a package _and_ missing off the extension is probably quite low?